src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -0,0 +1,421 @@
+
+                writer.WriteLine("return commands;");
+            }
+        }
+
+        protected virtual void WriteCommandHandlerBody(CodeMethod codeElement, RequestParams requestParams, bool isVoid, string returnType, LanguageWriter writer)
+        {
+            if (codeElement.HttpMethod == null) throw new InvalidOperationException("http method cannot be null");
+
+            var generatorMethod = (codeElement.Parent as CodeClass)
+                                                .Methods
+                                                .FirstOrDefault(x => x.IsOfKind(CodeMethodKind.RequestGenerator) && x.HttpMethod == codeElement.HttpMethod);
+            var requestBodyParam = requestParams.requestBody;
+            if (requestBodyParam != null)
+            {
+                var requestBodyParamType = requestBodyParam?.Type as CodeType;
+                if (requestBodyParamType?.TypeDefinition is CodeClass)
+                {
+                    writer.WriteLine($"using var stream = new MemoryStream(Encoding.UTF8.GetBytes({requestBodyParam.Name}));");
+                    writer.WriteLine($"var parseNode = ParseNodeFactoryRegistry.DefaultInstance.GetRootParseNode(\"{generatorMethod.ContentType}\", stream);");
+
+                    var typeString = conventions.GetTypeString(requestBodyParamType, requestBodyParam, false);
+
+                    if (requestBodyParamType.IsCollection)
+                    {
+                        writer.WriteLine($"var model = parseNode.GetCollectionOfObjectValues<{typeString}>();");
+                    }
+                    else
+                    {
+                        writer.WriteLine($"var model = parseNode.GetObjectValue<{typeString}>();");
+                    }
+
+                    requestBodyParam.Name = "model";
+                }
+                else if (conventions.StreamTypeName.Equals(requestBodyParamType?.Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    var name = requestBodyParam.Name;
+                    requestBodyParam.Name = "stream";
+                    writer.WriteLine($"using var {requestBodyParam.Name} = {name}.OpenRead();");
+                }
+            }
+
+            var parametersList = new CodeParameter[] { requestParams.requestBody, requestParams.queryString, requestParams.headers, requestParams.options }
+                                .Select(x => x?.Name).Where(x => x != null).DefaultIfEmpty().Aggregate((x, y) => $"{x}, {y}");
+            var separator = string.IsNullOrWhiteSpace(parametersList) ? "" : ", ";
+            WriteRequestInformation(writer, generatorMethod, parametersList, separator);
+
+            var requestMethod = "SendPrimitiveAsync<Stream>";
+            if (isVoid)
+            {
+                requestMethod = "SendNoContentAsync";
+            }
+
+            writer.WriteLine($"{(isVoid ? string.Empty : "var response = ")}await RequestAdapter.{requestMethod}(requestInfo);");
```

Comment:
I'd be nice to pass in the error mappings here so you'd benefit from the recent changes made in #1100 
https://github.com/microsoft/kiota/blob/1457eed0a80274466b01a77d88f574b77229cdf1/abstractions/dotnet/src/IRequestAdapter.cs#L54
If you want to test it, the openAPI description are not up to date yet, but I made a sample one here https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/1060

I'd be ok with logging that as a separate issue and coming back to it later if you want to merge this in first.